### PR TITLE
test(e2e): migrated tests to mocha/cypress, added helpers for applications vs. appeals

### PR DESCRIPTION
**Pull request by ob-pins**
## Describe your changes
Migrated cucumber tests to mocha/cypress. 
Not all files have been moved. These are 2 test suites still to be migrated.
Also added functionality for sharing test framework with appeals back office

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Review by beejjacobs — CHANGES_REQUESTED** [review]
Generally looks clear enough. A few minor comments with some changes to make.

File: apps/e2e/package.json [hunk 1]
```diff
@@ -5,8 +5,7 @@
 	"main": "index.js",
 	"scripts": {
 		"cy:open": "npx cypress open",
-		"cy:run": "npx cypress run",
-		"cy:run:smoke": "npx cypress run -e TAGS='@smoke'"
+		"cy:run": "npx cypress run"
```
[the remark this answers is not in the document]
Comment (ob-pins) [reply apps/e2e/package.json:8]: updated

File: apps/e2e/cypress/page_objects/uploadFiles.js [hunk 2]
```diff
@@ -6,9 +6,50 @@ export class FileUploadPage extends Page {
...
+		checkUpload();
+		checkUpload();
+		cy.contains('a', 'Download').should('exist');
+	}
+
+	downloadFile(fileNumber) {
+		cy.window()
+			.document()
+			.then(function (doc) {
+				doc.addEventListener('click', () => {
```
[the remark this answers is not in the document]
Comment (ob-pins) [reply apps/e2e/cypress/page_objects/uploadFiles.js:51]: done

File: apps/e2e/cypress/page_objects/createCasePage.js [hunk 3]
```diff
@@ -43,33 +43,41 @@ export class CreateCasePage extends Page {
...
+		}
 		this.sections.keyDates.fillSumbissionPublishedDate(projectInformation.publishedDate);
 		this.sections.keyDates.fillInternalAnticipatedDay(projectInformation.internalDateDay);
 		this.sections.keyDates.fillInternalAnticipatedMonth(projectInformation.internalDateMonth);
 		this.sections.keyDates.fillInternalAnticipatedYear(projectInformation.internalDateYear);
 		this.clickSaveAndContinue();
+		this.sections.checkYourAnswers.checkAllAnswers(projectInformation, mandatoryOnly);
 		this.clickButtonByText('I accept - confirm creation of a new case');
 		this.sections.caseCreated.validateCaseCreated();
+		cy.wait(2000);
```
[the remark this answers is not in the document]
Comment (ob-pins) [reply apps/e2e/cypress/page_objects/createCasePage.js]: removed

File: apps/e2e/cypress.config.js [hunk 4]
```diff
@@ -6,26 +6,24 @@ const {
...
-	deleteFile,
 	deleteDownloads,
-	webpackConfig
+	deleteUnwantedFixtures
 } = require('./cypress/support/cypressUtils');
-const preprocessor = require('@badeball/cypress-cucumber-preprocessor');
+const { getSpecPattern } = require('./cypress/support/utils/getSpecPattern');
 
 require('dotenv').config();
+const app = process.env.APP;
```
[the remark this answers is not in the document]
Comment (ob-pins) [reply apps/e2e/cypress.config.js:15]: done

**Review by beejjacobs — APPROVED** [review]

**Pull request closed without merging**